frontend/src/employee-mobile-frontend/staff-attendance/StaffAttendanceEditPage.tsx

```diff
@@ -0,0 +1,423 @@
+  groups: GroupInfo[],
+  attendance: StaffMemberAttendance | StaffAttendanceUpsert
+): StateOf<typeof staffAttendanceRow> => ({
+  id: attendance.id,
+  type: {
+    domValue: attendance.type,
+    options: staffAttendanceTypes.map((type) => ({
+      domValue: type,
+      value: type,
+      label: i18n.attendances.staffTypes[type]
+    }))
+  },
+  groupId: {
+    domValue: attendance.groupId || '',
+    options: [
+      { domValue: '', value: null, label: i18n.attendances.noGroup },
+      ...groups.map((group) => ({
+        domValue: group.id,
+        value: group.id,
+        label: group.name
+      }))
+    ]
+  },
+  arrivedDate: localDate.fromDate(attendance.arrived.toLocalDate()),
+  arrivedTime: attendance.arrived.toLocalTime().format(),
+  departedDate: localDate.fromDate(attendance.departed?.toLocalDate() ?? date),
+  departedTime: attendance.departed?.toLocalTime().format() ?? ''
+})
+
+export default React.memo(function StaffAttendanceEditPage() {
+  const { unitId, employeeId } = useNonNullableParams<{
+    unitId: string
+    employeeId: string
+  }>()
+  const { i18n } = useTranslation()
+  const { unitInfoResponse } = useContext(UnitContext)
+  const staffAttendanceResponse = useQueryResult(staffAttendanceQuery(unitId))
+  const combinedResult = useMemo(
+    () =>
+      combine(unitInfoResponse, staffAttendanceResponse).map(
+        ([{ groups, staff }, { staff: staffMember }]) => ({
+          groups: sortBy(groups, (group) => group.name),
+          staff: staff.find((s) => s.id === employeeId),
+          staffMember: staffMember.find((s) => s.employeeId === employeeId)
+        })
+      ),
+    [employeeId, unitInfoResponse, staffAttendanceResponse]
+  )
+
+  return renderResult(combinedResult, ({ groups, staff, staffMember }) => (
+    <StaffMemberPageContainer>
+      {staffMember === undefined || staff === undefined ? (
+        <ErrorSegment title={i18n.attendances.staff.errors.employeeNotFound} />
+      ) : !staff.pinSet ? (
+        <ErrorSegment title={i18n.attendances.staff.pinNotSet} />
+      ) : staff.pinLocked ? (
+        <ErrorSegment title={i18n.attendances.staff.pinLocked} />
+      ) : (
+        <StaffAttendanceEditor
+          unitId={unitId}
+          employeeId={employeeId}
+          groups={groups}
+          staffMember={staffMember}
+        />
+      )}
+    </StaffMemberPageContainer>
+  ))
+})
+
+const StaffAttendanceEditor = ({
+  unitId,
+  employeeId,
+  groups,
+  staffMember
+}: {
+  unitId: UUID
+  employeeId: UUID
+  groups: GroupInfo[]
+  staffMember: StaffMember
+}) => {
+  const { groupRoute } = useSelectedGroup()
+  const { i18n, lang } = useTranslation()
+  const navigate = useNavigate()
+  const [date] = useState(LocalDate.todayInHelsinkiTz())
+  const form = useForm(
+    staffAttendancesForm,
+    () => initialFormState(i18n, date, groups, staffMember.attendances),
+    i18n.attendances.staff.validationErrors
+  )
+  const { pinCode, rows } = useFormFields(form)
```

Comment:
Use `const boundRows = useFormElems(rows)` here to avoid `useFormElem` + if + throw in the `StaffAttendanceRowEditor` component. The prop's type will be `bind: BoundForm<typeof staffAttendanceRow>`.

Reply:
Fixed